sql/item.cc

```diff
@@ -2176,6 +2176,11 @@ Item::Type Item_name_const::type() const
 
 bool Item_name_const::fix_fields(THD *thd, Item **ref)
 {
+  if (thd->where == THD_WHERE::WHERE_CLAUSE) {
+      value_item->fix_fields_if_needed(thd, &value_item);
+      thd->change_item_tree(ref, value_item);
```

Comment:
This needs error handling. I think one needs t ocall fix_fields_if_needed for both value_item and name_item and check the result.

I'm wondering why this is needed:

```cpp
 thd->change_item_tree(ref, value_item);
```
did you have anything in mind?

Reply:
Without the call to `change_item_tree`, the node for `NAME_CONST` remains in the parse tree.  This call replaces the `NAME_CONST` node with the value it wraps.